scapy/pton_ntop.py

```diff
@@ -75,15 +83,7 @@ def inet_ntop(af, addr):
                 hexstr = hex(value)[2:]
             except TypeError:
                 raise Exception("Illegal syntax for IP address")
-            parts.append(hexstr.lstrip("0").lower())
-        result = ":".join(parts)
-        while ":::" in result:
-            result = result.replace(":::", "::")
-        # Leaving out leading and trailing zeros is only allowed with ::
-        if result.endswith(":") and not result.endswith("::"):
-            result = result + "0"
-        if result.startswith(":") and not result.startswith("::"):
-            result = "0" + result
-        return result
-    else:
-        raise Exception("Address family not supported yet")        
+            parts.append(hexstr.lower())
+
+        # Note: the returned address is never compact
+        return ":".join(parts)
```

Comment:
I tried to work with regex. I think I found a way to compact the IP again.
```python
import re

ips = [
    '0:0:1234:0:0:abcd:1000:0123:0:0:0:0:5678:0:0:0:0',
    '0:0:1234:0:0:0:abcd:1000:0123:0:0:0:0:5678:0:0:0',
    '0:0:123:45:1:7894:1234',
    '0:0:123:45:1:7894:1234:0:0',
    '123:45:1:7894:1234:0:0',
    '123:45:1:7894:1234:0:123',
    '0:0:0:0:0:0:0:0:0:1',
    '1:0:0:0:0:0:0:0:0:0',
    '0:0:0:0:0:0:0:0:0:0',
    '0:0:0:0:0:0:0:1:0:1',
    '0:1:0:1:0:1:0:1:0:1',
    '1:0:1:0:1:0:1:0:1:0',
]

for ip in ips:
    matches = re.findall('(?::|^)(0(?::0)*)(?::|$)', ip)
    match = max(matches)
    leftidx = ip.rfind(match)
    left = ip[:leftidx]
    rightidx = leftidx + len(match)
    if len(ip) == rightidx:
        result = left + ":"
    elif leftidx == 0:
        result = ":" + ip[rightidx:]
    else:
        result = left + ip[rightidx:]

    print("{} -> {}".format(ip,result))
```
The only think not working correctly is the undefined IP (0:0:...:0), which should not occur anyways?